TailLight/device.cpp

```diff
@@ -70,6 +95,62 @@ static UNICODE_STRING GetDevicePropertyString(WDFDEVICE device, DEVICE_REGISTRY_
     return result;
 }
 
+static VOID WdfMemoryToUnicodeString(WDFMEMORY memory, UNICODE_STRING& string)
+/*++
+Routine Description:
+    EvtDriverDeviceAdd is called by the framework in response to AddDevice
+    call from the PnP manager. We create and initialize a device object to
+    represent to be part of the device stack as a filter.
+
+Arguments:
+    Driver - Handle to a framework driver object created in DriverEntry
+
+    DeviceInit - Pointer to a framework-allocated WDFDEVICE_INIT structure.
+--*/
+{
+    // initialize string based on memory
+    size_t bufferLength = 0;
+    string.Buffer = (WCHAR*)WdfMemoryGetBuffer(memory, &bufferLength);
+    if (string.Buffer == NULL)
+        return;
+
+    string.MaximumLength = (USHORT)bufferLength;
+    string.Length = (USHORT)bufferLength - sizeof(UNICODE_NULL);
+}
+
+UNICODE_STRING GetDevicePropertyString(WDFIOTARGET target, 
```

Comment:
It seems like you've introduced a second `GetDevicePropertyString` function that's almost identical to the existing `GetDevicePropertyString` function, with the exception of a `WDFIOTARGET` argument. Can you please try to modify/extend the existing function instead of introducing a semi-duplicated function?

Reply:
I've fixed using a pointer-to-function approach. This avoids vtables.

Reply:
https://github.com/forderud/IntelliMouseDriver/pull/54/files#r1518576669